mlflow/utils/rest_utils.py

```diff
@@ -169,6 +171,48 @@ def call_endpoint(host_creds, endpoint, method, json_body, response_proto):
     return response_proto
 
 
+# Response codes that generally indicate transient network failures and merit client retries,
+# based on guidance from cloud service providers
+# (https://docs.microsoft.com/en-us/azure/architecture/best-practices/retry-service-specific#general-rest-and-retry-guidelines)
+TRANSIENT_FAILURE_RESPONSE_CODES = [
+    408, # Request Timeout
+    429, # Too Many Requests
+    500, # Internal Server Error
+    502, # Bad Gateway
+    503, # Service Unavailable
+    504, # Gateway Timeout
+]
+
+
+def http_put(*args, **kwargs):
```

Comment:
Cool! Can we include & leverage a similar `http_get` method to provide retries for AWS / Azure / GCP uploads & downloads (https://github.com/mlflow/mlflow/blob/ef17afd5766dff33cf2c8d963223eb7be4a61b61/mlflow/store/artifact/databricks_artifact_repo.py#L194-L206, https://github.com/mlflow/mlflow/blob/ef17afd5766dff33cf2c8d963223eb7be4a61b61/mlflow/store/artifact/databricks_artifact_repo.py#L244)?